iothub/device/tests/Transport/AmqpIot/AmqpIotErrorAdapterTests.cs

```diff
@@ -0,0 +1,160 @@
+        
+        public void AmqpIotErrorAdapter_ToIotHubClientContract_AmqpException(string amqpErrorCode, IotHubClientErrorCode iotHubClientErrorCode)
+        {
+            var error = new Error
+            {
+                Condition = amqpErrorCode,
+            };
+            IotHubClientException ex = AmqpIotErrorAdapter.ToIotHubClientContract(new AmqpException(error));
+            ex.ErrorCode.Should().Be(iotHubClientErrorCode);
+        }
+
+        [TestMethod]
+        [DataRow("amqp:resource-limit-exceeded", IotHubClientErrorCode.QuotaExceeded)]
+        [DataRow("amqp:link:message-size-exceeded", IotHubClientErrorCode.MessageTooLarge)]
+        [DataRow("amqp:unauthorized-access", IotHubClientErrorCode.Unauthorized)]
+        [DataRow("amqp:not-found", IotHubClientErrorCode.DeviceNotFound)]
+        [DataRow(AmqpIotConstants.Vendor + ":timeout", IotHubClientErrorCode.Timeout)]
+        [DataRow(AmqpIotConstants.Vendor + ":message-lock-lost", IotHubClientErrorCode.DeviceMessageLockLost)]
+        [DataRow(AmqpIotConstants.Vendor + ":device-container-throttled", IotHubClientErrorCode.Throttled)]
+        [DataRow(AmqpIotConstants.Vendor + ":iot-hub-suspended", IotHubClientErrorCode.Suspended)]
+        public void AmqpIotErrorAdapter_ToIotHubClientContract_Error(string amqpErrorCode, IotHubClientErrorCode iotHubClientErrorCode)
+        {
+            var error = new Error
+            {
+                Condition = amqpErrorCode,
+            };
+            IotHubClientException ex = AmqpIotErrorAdapter.ToIotHubClientContract(error);
+            ex.ErrorCode.Should().Be(iotHubClientErrorCode);
+        }
+
+        [TestMethod]
+        [DataRow("amqp:resource-locked")]
+        [DataRow("amqp:connection:forced")]
+        [DataRow("amqp:connection:framing-error")]
+        [DataRow("amqp:connection:redirect")]
+        [DataRow("amqp:session:window-violation")]
+        [DataRow("amqp:session-errant-link")]
+        [DataRow("amqp:session:handle-in-use")]
+        [DataRow("amqp:session:unattached-handle")]
+        [DataRow("amqp:link:detach-forced")]
+        [DataRow("amqp:link:transfer-limit-exceeded")]
+        [DataRow("amqp:link:redirect")]
+        [DataRow("amqp:link:stolen")]
+        public void AmqpIotErrorAdapter_ToIotHubClientContract_IsTransient(string amqpErrorCode)
+        {
+            var error = new Error
+            {
+                Condition = amqpErrorCode,
+            };
+            IotHubClientException ex = AmqpIotErrorAdapter.ToIotHubClientContract(new AmqpException(error));
+            ex.IsTransient.Should().BeTrue();
+        }
+
+        [TestMethod]
+        public void AmqpIotErrorAdapter_ToIotHubClientContract_Error_Null()
+        {
+            IotHubClientException ex = AmqpIotErrorAdapter.ToIotHubClientContract((Error) null);
+            ex.Should().BeEquivalentTo(new IotHubClientException("Unknown error."));
```

Comment:
Instead of comparing string, how about we compare ex.IotHubClientErrorCode is IotHubClientErrorCode.Unknown?